### ISSUE TITLE
References in rst files break with modern Sphinx

### ISSUE THREAD

**Issue by jmlarson1** [Libensemble/libensemble#256]
If we require a modern version of sphinx, the references in `libensemble/docs/tutorials/local_sine_tutorial.rst` 
(and elsewhere) no longer work:

See the latexmk tab here:
https://readthedocs.org/projects/libensemble/builds/9744900/

And the PDF does not have working hyperlinks. But the same thing happened before, but the PDF links worked:
https://readthedocs.org/projects/libensemble/builds/9721766/

**Comment by jlnav** [comment]
I also found the same `LaTeX Warning: Hyper reference 'tutorials/local_sine_tutorial::doc' on page 1 undefined on input line 120.` line on the first build that includes the tutorial:

https://readthedocs.org/projects/libensemble/builds/9332076/

Locally, I was able to build the HTML with Sphinx 2.2.0 with no reference issues. I haven't tried a local latexmk yet.

Although the hyperlinks are all warning as undefined, which ones are failing in practice?
